### PR TITLE
fix ant-design/ant-design#21161 Select can not clear when value change to `null`

### PR DESCRIPTION
**Pull request by morenyang**
try to fix https://github.com/ant-design/ant-design/issues/21161

### CODE REVIEW

File: src/generate.tsx [hunk 1]
```diff
@@ -374,8 +374,7 @@ export default function generateSelector<
     const [innerValue, setInnerValue] = React.useState<ValueType>(
       value || defaultValue,
     );
-    const baseValue =
-      value !== undefined && value !== null ? value : innerValue;
+    const baseValue = 'value' in props ? value : innerValue;
```
Comment (yoyo837) [inline src/generate.tsx]: ref: ant-design/ant-design#20979 #443 You can not fix it in this way!

Comment (morenyang) [reply src/generate.tsx]: maybe my understanding of this code was wrong...

Comment (yoyo837) [reply src/generate.tsx]: I think we should delete `&& value !== null` to fix it.

Comment (yoyo837) [reply src/generate.tsx]: @zombieJ

File: tests/Select.test.tsx [hunk 2]
```diff
@@ -1458,6 +1458,15 @@ describe('Select.Basic', () => {
     expect(wrapper.find('.rc-select-selection-item')).toHaveLength(0);
   });
 
+  it('reset value to null should reset display value', () => {
+    const wrapper = mount(<Select value="light" />);
+    expect(wrapper.find('.rc-select-selection-item').text()).toEqual('light');
+
+    wrapper.setProps({ value: null });
```
Comment (yoyo837) [inline tests/Select.test.tsx]: https://github.com/ant-design/ant-design/issues/20979#issuecomment-575896823
`undefined` is uncontrolled, but null is not.

Comment (morenyang) [reply tests/Select.test.tsx]: emmm, 
agree with you

Comment (morenyang) [reply tests/Select.test.tsx]: bug set to `null` can clear the select in 3.x

Comment (yoyo837) [reply tests/Select.test.tsx]: I'm not sure about this in 3.x, I need to check the code the day after tomorrow to understand it. Maybe someone else knew this before me.

Comment (morenyang) [reply tests/Select.test.tsx]: https://codesandbox.io/s/antd-select-clear-0yurp

emmm
seems in 3.x, set to `null` is not a **clear** action

**Comment by morenyang** [comment]
Close this PR because change value to `null` should not trigger **clear** action.

**Pull request closed without merging**